applications/vanilla/models/class.categorymodel.php

```diff
@@ -1736,7 +1736,9 @@ public static function joinRecentChildPosts(&$category = null, &$categories = nu
      * @param bool $addUserCategory
      */
     public static function joinUserData(&$categories, $addUserCategory = true) {
-        $iDs = array_keys($categories);
+        $iDs = array_column($categories, 'CategoryID', 'CategoryID');
+        $categoriesKeys = array_column($categories, 'CategoryID');
```

Comment:
This looks like an extra line of code you can avoid.

```
$iDs = array_column($categories, 'CategoryID', 'CategoryID');
$categories = array_combine($iDs, $categories);
```

Reply:
I don't really follow why we extract the ID's.  But that being said, I don't I think doing a refactor on the code increases the risk of breaking something else.  I'm not sure the if there is any value doing a refactor in the code.